### PR TITLE
Untested: Have Autorevision create a PR instead of trying to push

### LINKED ISSUE #955: AutoRevision.txt CI update fails at release time
**Describe the bug**
The CI to update AutoRevision.txt as part of the packaging of source archive files fails with the following error message:

```
remote: error: GH006: Protected branch update failed for refs/heads/master.        
remote: error: 7 of 7 required status checks are expected. Changes must be made through a pull request.        
To https://github.com/longturn/freeciv21
 ! [remote rejected]       master -> master (protected branch hook declined)
error: failed to push some refs to 'https://github.com/longturn/freeciv21'
Error: Invalid status code: 1
    at ChildProcess.<anonymous> (/home/runner/work/_actions/stefanzweifel/git-auto-commit-action/v4/index.js:17:19)
    at ChildProcess.emit (events.js:314:20)
    at maybeClose (internal/child_process.js:1022:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:287:5) {
  code: 1
}
Error: Invalid status code: 1
    at ChildProcess.<anonymous> (/home/runner/work/_actions/stefanzweifel/git-auto-commit-action/v4/index.js:17:19)
    at ChildProcess.emit (events.js:314:20)
    at maybeClose (internal/child_process.js:1022:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:287:5)
```

**To Reproduce**
1. Create a new Release and set an updated tag ID
2. Publish the release
3. See error

**Expected behavior**
The CI should be able to update the given file and produce a ZIP and TAR.GZ archive of the code at the referenced tag.

**Screenshots**
N/A

**Platform and version (please complete the following information):**
 - OS: N/A
 - Freeciv21 version: 3.0-beta.1
 - Ruleset/Longturn game (if applicable): N/A

**Additional context**
None

### PR DESCRIPTION
**Pull request by lmoureaux**
Direct pushes to master are forbidden by branch protection rules. Have the
automation create a PR instead, so we can approve it and enable automerge. It's
a bit less convenient than a push but good enough for now. It also lets us
check that the action is working is intended. If this works well, we could
consider enabling automerge from within the action (or pushing directly as it
looks like it's supported via an access token [1]).

Closes #955.

[1] https://github.com/stefanzweifel/git-auto-commit-action/issues/87

### CODE REVIEW

**Comment by jwrober** [comment]
Does this mean that when we tag a release, the whole release process is paused until we approve of the PR and then once approved all the rest of the release process will continue including making the `.zip`, `.gz` archives and the varying binary packages for Windows, Debian and MacOS?

**Comment by lmoureaux** [comment]
> Does this mean that when we tag a release, the whole release process is paused until we approve of the PR and then once approved all the rest of the release process will continue including making the `.zip`, `.gz` archives and the varying binary packages for Windows, Debian and MacOS?

Source archive creation is not paused. I guess we should also run the autorevision updater before creating the tarball? Binary packages rely on the `git` version of autorevision.

**Review by jwrober — APPROVED** [review]

**Pull request merged**
